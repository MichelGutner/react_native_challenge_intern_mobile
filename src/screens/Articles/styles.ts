import styled from 'styled-components/native';

export const Container = styled.SafeAreaView`
  flex: 1;
`;

export const ArticlesLenght = styled.Text`
  font-size: 16px;
  color: #fff;
`;
