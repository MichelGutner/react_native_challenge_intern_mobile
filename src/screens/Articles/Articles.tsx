import React, { useEffect, useState } from 'react';
import { Alert, FlatList } from 'react-native';
import { Header } from 'react-native-elements';
import { Themes } from '../../../themes/themes';
import ArticlesItem from '../../components/articlesItem/ArticlesItem';
import SearchArticles from '../../components/searchArticles/SearchArticles';
import { getArticles } from '../../services/healthCareApi';
import { ArticlesLenght, Container } from './styles';

const renderItem = ({ item }) => <ArticlesItem item={item} />;

const Articles = () => {
  const [articles, setArticles] = useState([]);
  const [searchArticles, setSearchArticles] = useState('');

  useEffect(() => {
    getArticles(
      json => {
        setArticles(json.articles);
      },
      () => {
        Alert.alert('Algum erro aconteceu');
      }
    );
  }, []);

  const articlesOrder = [...articles];
  articlesOrder.sort((ordenationPrev, OrdernationNext) =>
    ordenationPrev.date < OrdernationNext.date
      ? 1
      : OrdernationNext.date < ordenationPrev.date
      ? -1
      : 0
  );

  useEffect(() => {
    if (searchArticles === '') {
      getArticles(
        json => {
          setArticles(json.articles);
        },
        () => {
          Alert.alert('Algum erro aconteceu');
        }
      );
    } else {
      setArticles(
        articlesOrder.filter(item => {
          if (
            item.title.toLowerCase().indexOf(searchArticles.toLowerCase()) > -1
          ) {
            return true;
          } else {
            return false;
          }
        })
      );
    }
  }, [searchArticles]);

  const articlesLength = articles.length;

  return (
    <Container>
      <Header
        containerStyle={{
          backgroundColor: Themes.colors.backgroundSanar,
        }}
        placement="left"
        leftComponent={
          <ArticlesLenght>
            {articlesLength + ' Artigos encontrado'}
          </ArticlesLenght>
        }
        centerComponent={
          <SearchArticles
            value={searchArticles}
            onChangeText={(text: string) => setSearchArticles(text)}
          />
        }
        rightComponent={{
          icon: 'search',
          color: '#fff',
        }}
      />
      <FlatList
        data={articlesOrder}
        keyExtractor={item => item.title.toString()}
        renderItem={renderItem}
      />
    </Container>
  );
};

export default Articles;
