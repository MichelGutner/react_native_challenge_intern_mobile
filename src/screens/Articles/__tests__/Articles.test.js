import React from 'react';
import renderer from 'react-test-renderer';
import Articles from '../Articles';

test('renders correctly', () => {
  const tree = renderer.create(<Articles />).toJSON();
  expect(tree).toMatchSnapshot();
});
