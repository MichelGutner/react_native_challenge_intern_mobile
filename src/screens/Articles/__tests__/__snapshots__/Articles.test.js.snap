// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Array [
      Object {
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#0050F0",
        "borderBottomColor": "#f2f2f2",
        "borderBottomWidth": 0.5,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingHorizontal": 10,
        "paddingVertical": 10,
      }
    }
    testID="headerContainer"
  >
    <RNCSafeAreaView
      edges={
        Array [
          "left",
          "top",
          "right",
        ]
      }
      style={
        Object {
          "flexDirection": "row",
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "flex-start",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#fff",
                "fontSize": 16,
              },
            ]
          }
        >
          0 Artigos encontrado
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "flex": 3,
            "paddingHorizontal": 15,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgba(255,255,255, 0.08)",
                "borderRadius": 20,
                "height": 25,
                "justifyContent": "center",
                "width": 140,
              },
            ]
          }
        >
          <TextInput
            maxLength={15}
            onChangeText={[Function]}
            placeholder="Buscar..."
            placeholderTextColor="#FFF"
            style={
              Array [
                Object {
                  "color": "#FFF",
                  "fontSize": 13,
                  "paddingBottom": 2,
                  "paddingLeft": 20,
                },
              ]
            }
            value=""
          />
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "flex-end",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "flex-end",
              "overflow": "hidden",
            }
          }
        >
          <View
            icon="search"
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "justifyContent": "center",
                }
              }
            >
              <Text
                allowFontScaling={false}
                selectable={false}
                style={
                  Array [
                    Object {
                      "color": "#fff",
                      "fontSize": 24,
                    },
                    Object {
                      "backgroundColor": "transparent",
                    },
                    Object {
                      "fontFamily": "Material Icons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
                testID="iconIcon"
              >
                î¢¶
              </Text>
            </View>
          </View>
        </View>
      </View>
    </RNCSafeAreaView>
  </View>
  <RCTScrollView
    data={Array []}
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    viewabilityConfigCallbackPairs={Array []}
  >
    <View />
  </RCTScrollView>
</RCTSafeAreaView>
`;
